### PR TITLE
Control host deletion using env variable

### PR DESCRIPTION
**Pull request by sijeesh**

### CODE REVIEW

File: helm/charts/hpe-csi-driver/README.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ The following table lists the configurable parameters of the chart and their def
 | disableNodeConfiguration  | Disables node conformance and configuration.`*`                        | false            |
 | disableNodeGetVolumeStats | Disable NodeGetVolumeStats call to CSI driver.                         | false            |
 | disableNodeMonitor        | Disables the Node Monitor that manages stale storage resources.        | false            |
+| disableHostDeletion       | Disables host deletion during volume unpublish                         | false            |
```
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/README.md:41]: Can you change this to `Disables host deletion by the CSP when no volumes are associated with the host.`? "volume unpublish" is an implementation detail users won't care about. Please change throughout the PR.

**Review by datamattsson — CHANGES_REQUESTED** [review]
Nits.

File: helm/charts/hpe-csi-driver/README.md (hunk 1 above)
Comment (sijeesh) [reply helm/charts/hpe-csi-driver/README.md]: Done

File: helm/charts/hpe-csi-driver/README.md [hunk 2]
```diff
@@ -27,38 +27,38 @@ Depending on which [Container Storage Provider](https://scod.hpedev.io/container
...
+| disable.primera           | Disable HPE Primera (and 3PAR) CSP `Service`.                                  | false            |
+| disable.alletra6000       | Disable HPE Alletra 5000/6000 CSP `Service`.                                   | false            |
+| disable.alletra9000       | Disable HPE Alletra 9000 CSP `Service`.                                        | false            |
+| disable.alletraStorageMP  | Disable HPE Alletra Storage MP CSP `Service`.                                  | false            |
+| disableNodeConformance    | Disable automatic installation of iSCSI, multipath and NFS packages.           | false            |
+| disableNodeConfiguration  | Disables node conformance and configuration.`*`                                | false            |
+| disableNodeGetVolumeStats | Disable NodeGetVolumeStats call to CSI driver.                                 | false            |
+| disableNodeMonitor        | Disables the Node Monitor that manages stale storage resources.                | false            |
+| disableHostDeletion       | Disables host deletion by the CSP when no volumes are associated with the host | false            |
```
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/README.md:41]: Can you make sure to follow the punctuation patterns throughout the PR as well?

File: helm/charts/hpe-csi-driver/templates/nimble-csp.yaml [hunk 3]
```diff
@@ -65,6 +65,11 @@ spec:
           resources:
             {{- toYaml .Values.csp.resources | nindent 12 }}
           imagePullPolicy: {{ .Values.imagePullPolicy | quote }}
+          env:
+            {{- if .Values.disableHostDeletion }}
+            - name: DISABLE_HOST_DELETION
+              value: {{ .Values.disableHostDeletion }}
```
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/templates/nimble-csp.yaml:71]: Environment variables are strings. Please follow the pattern for previous `disable*` parameters by setting `- value: "true"`.

```
json: cannot unmarshal bool into Go struct field EnvVar.spec.template.spec.containers.env.value of type string
```

File: helm/charts/hpe-csi-driver/templates/primera-3par-csp.yaml [hunk 4]
```diff
@@ -97,6 +97,10 @@ spec:
             - name: CHAP_SECRET_NAMESPACE
               value: {{ .Release.Namespace }}
             {{- end }}
+            {{- if .Values.disableHostDeletion }}
+            - name: DISABLE_HOST_DELETION
+              value: {{ .Values.disableHostDeletion }}
```
Comment (datamattsson) [inline helm/charts/hpe-csi-driver/templates/primera-3par-csp.yaml:102]: Same as comments above.

**Review by datamattsson — CHANGES_REQUESTED** [review]
Nits.

**Comment by sijeesh** [comment]
@datamattsson  , addressed all the review comments . Please review.

File: helm/charts/hpe-csi-driver/templates/primera-3par-csp.yaml (hunk 4 above)
Comment (sijeesh) [reply helm/charts/hpe-csi-driver/templates/primera-3par-csp.yaml]: Updated

File: helm/charts/hpe-csi-driver/README.md (hunk 2 above)
Comment (sijeesh) [reply helm/charts/hpe-csi-driver/README.md]: Updated to match the pattern we followed in each file.

**Review by datamattsson — COMMENTED** [review]
Why did we remove the Operator chunks?

**Comment by sijeesh** [comment]
> Why did we remove the Operator chunks?

it was the from the destination directories ? do we need to keep that? referred tearlier PRs for the same

**Comment by datamattsson** [comment]
> > Why did we remove the Operator chunks?
> 
> it was the from the destination directories ? do we need to keep that? referred earlier PRs for the same

Ok, I can see that now. LGTM.

**Review by datamattsson — APPROVED** [review]

**Review by dileepds — APPROVED** [review]
LGTM

**Pull request merged**